### DISCUSSION TITLE
Questions when using CBC

### DISCUSSION THREAD

**Discussion by CruiseSun1998** [yalmip/YALMIP#1168] [Problems using YALMIP] [answered]
Dear Professor:

Recently I am using Yalmip together with CBC (in the OPTI Toolbox) to solve MIP problems. And I have a question:

============================================================
*** MAXIMUM TIME REACHED ***
 Objective Value: -1.09833e+07
 Gap:             538844 (4.91%)
 Searched:        11761 nodes [214487 LP iterations]

------------------------------------------------------------------

    yalmipversion: '20210331'
    matlabversion: '9.5.0.1178774 (R2018b) Update 5'
       yalmiptime: 0.1350
       solvertime: 999.8200
             info: 'Maximum iterations or time limit exceeded (CBC)'
          problem: 3
============================================================

I failed to solve an MILP problem due to the maximum time.

However, I didn't find an option to change the parameter "maximum time" in ops.cbc .

I wonder how to change the parameters like "maximum time" and "maximum nodes" for CBC.

Best regards,
Cruise

**Comment by johanlofberg** [comment, marked as answer]
Opti has some mix of common and solver specific, so I think you have to do something like below to add some some missing options

ops = sdpsettings('solver','cbc');
ops.cbc.maxtime = 1;

YALMIP creates the options via cbcset, and some are not added by that constructor. You can look in opti_cbc to reverse engineer some stuff
